Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -2049,6 +2041,94 @@ final class BuildPlanTests: XCTestCase {
+        )
+        XCTAssertNoDiagnostics(observability.diagnostics)
+
+        let parameters = mockBuildParameters(config: .release, destinationTriple: .wasi)
+        let result = try BuildPlanResult(plan: BuildPlan(
+            buildParameters: parameters,
+            graph: graph,
+            fileSystem: fs,
+            observabilityScope: observability.topScope
+        ))
+        result.checkProductsCount(2)
+        result.checkTargetsCount(4)
+
+        let lib = try result.target(for: "lib").clangTarget()
+        let args = [
+            "-target", "wasm32-unknown-wasi", "-g", "-O2", "-DSWIFT_PACKAGE=1",
+            "-fblocks", "-fmodules", "-fmodule-name=lib", "-I", "/Pkg/Sources/lib/include",
+            "-fmodules-cache-path=/path/to/build/release/ModuleCache"
+        ]
+        XCTAssertEqual(try lib.basicArguments(isCXX: false), args)
+        XCTAssertEqual(lib.objects, [AbsolutePath("/path/to/build/release/lib.build/lib.c.o")])
+        XCTAssertEqual(lib.moduleMap, AbsolutePath("/path/to/build/release/lib.build/module.modulemap"))
+
+        let exe = try result.target(for: "app").swiftTarget().compileArguments()
+        XCTAssertMatch(
+            exe,
+            [
+                "-swift-version", "4", "-O", "-g", .equal(j),
+                "-DSWIFT_PACKAGE", "-Xcc", "-fmodule-map-file=/path/to/build/release/lib.build/module.modulemap",
+                "-Xcc", "-I", "-Xcc", "/Pkg/Sources/lib/include",
+                "-module-cache-path", "/path/to/build/release/ModuleCache", .anySequence
+            ]
+        )
+
+        let appBuildDescription = try result.buildProduct(for: "app")
+        XCTAssertEqual(
+            try appBuildDescription.linkArguments(),
+            [
+                "/fake/path/to/swiftc", "-g", "-L", "/path/to/build/release",
+                "-o", "/path/to/build/release/app.wasm",
+                 "-module-name", "app", "-static-stdlib", "-emit-executable",
+                 "@/path/to/build/release/app.product/Objects.LinkFileList",
+                 "-target", "wasm32-unknown-wasi"
+            ]
+        )
+
+        let executablePathExtension = appBuildDescription.binary.extension
+        XCTAssertEqual(executablePathExtension, "wasm")
+
+        let testBuildDescription = try result.buildProduct(for: "PkgPackageTests")
+        XCTAssertEqual(
+            try testBuildDescription.linkArguments(),
+            [
+                "/fake/path/to/swiftc", "-g", "-L", "/path/to/build/release",
+                "-o", "/path/to/build/release/PkgPackageTests.wasm",
+                "-module-name", "PkgPackageTests", "-emit-executable",
+                "@/path/to/build/release/PkgPackageTests.product/Objects.LinkFileList",
+                "-target", "wasm32-unknown-wasi"
+            ]
+        )
+
+        let testPathExtension = testBuildDescription.binary.extension
+        XCTAssertEqual(testPathExtension, "wasm")
+    }
+
```

Comment:
@MaxDesiatov one of the key pieces of feedback to this proposal was that this should only apply in release mod (since if you are trying to debug you would need to have the toolchain installed by definition). 

ptal at the changes to the WASI tests: expanded them to cover both debug and release modes

Reply:
Makes sense, thanks!